src/exportJarSteps/ResolveMainMethodStep.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { ProgressLocation, QuickInputButtons, window } from "vscode";
+import { GenerateSettings } from "../exportJarFileCommand";
+import { Jdtls } from "../java/jdtls";
+import { ExportSteps, IStep } from "./IStep";
+import { createPickBox, IJarQuickPickItem } from "./utility";
+
+export class ResolveMainMethodStep implements IStep {
+
+    private static getName(data: MainMethodInfo) {
+        return data.name.substring(data.name.lastIndexOf(".") + 1);
+    }
+
+    public exportStep;
+
+    constructor() {
+        this.exportStep = ExportSteps.ResolveMainMethod;
+    }
+
+    public async execute(lastStep: IStep | undefined, generateSettings: GenerateSettings): Promise<ExportSteps> {
+        return await this.resolveMainMethod(lastStep, generateSettings) ? ExportSteps.GenerateJar : lastStep.exportStep;
```

Comment:
Confused here. Why we need to pass the lastStep into every execution, instead of directly resolve the next step enum?